### ISSUE TITLE
Extend "using-constant-test" to checks involving boolean operators

### ISSUE THREAD

**Issue by cfax** [pylint-dev/pylint#7933]
### Current problem

We have recently found a case in which Pylint could help.
Specifically, we had this check:

```python
if get_text() == "string A":
    pass
```

That we wanted to extend for "string B".
A bug was then introduced, in this  form:
```python
if get_text() == "string A" or "string B":
    pass
```
which always evaluate to True.

I tried to see whether "using-constant-text" could be extended to cater for this type of bugs, but I think it might not be that easy - or at all feasible.
In case it is, I'd be happy to contribute, but I guess I will need some pointers.

### Desired solution

Ideally a warning should appear to highlight the fact that the comparison is not what the author likely wants.

**Comment by crazybolillo** [comment]
I am interested in taking this. I looked through the message list and was not able to found something similar. Are you guys still interested in a PR for this?

**Comment by Pierre-Sassoulas** [comment]
Yes, we only need to define the message that would be raised. Do you have an opinion ?

**Comment by crazybolillo** [comment]
I was thinking something like:

`always-true-eval`

"Statement always evaluates as true"

**Comment by Pierre-Sassoulas** [comment]
I checked and I think there's [``condition-evals-to-constant``]( https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/condition-evals-to-constant.html) that is more generic.

**Comment by crazybolillo** [comment]
That is nice and confirms the deja vu feeling I had. It seems like the checker will just need some work to take into consideration other expressions, like the one reported in this issue.
